etc/errors/anvil.json

```diff
@@ -1058,6 +1145,55 @@
                 ]
               }
             },
+            {
+              "name": "TransactionGasEstimationFailed",
+              "code": 11,
+              "message": "Gas estimation failed: \n{inner}",
+              "fields": [
+                {
+                  "name": "inner",
+                  "type": "GasEstimation"
+                },
+                {
+                  "name": "transaction_data",
+                  "type": "bytes"
+                }
+              ],
+              "doc": {
+                "summary": "Transaction gas estimation failed in anvil-zksync.",
+                "description": [
+                  "This error occurs when a gas estimation for transaction failed.",
+                  "This is a wrapper error that contains a more specific gas estimation error inside it, which provides details about the cause of failure."
+                ],
+                "likely_causes": [
+                  {
+                    "cause": "The transaction always reverts, so it can't be estimated.",
+                    "fixes": [
+                      "Inspect the source code of your contracts and debug the transaction to eliminate the cause of revert."
+                    ]
+                  },
+                  {
+                    "cause": "The transaction always halts, so it can't be estimated.",
+                    "fixes": [
+                      "Inspect the source code of your contracts and debug the transaction to eliminate the cause of halt."
+                    ]
+                  },
+                  {
+                    "cause": "The transaction requires too much gas to be executed. For example, it may contain an infinite loop.",
+                    "fixes": [
+                      "Debug the transaction to see why it consumes more gas than is allowed."
+                    ]
+                  },
+                  {
+                    "cause": "The transaction is malformed.",
+                    "fixes": [
+                      "Check the nonce value provided in the RPC call that you are using (such as `eth_estimateGas`). If you do not care about the specific nonce for the transaction, do not provide any value for it.",
+                      "Check all arguments of the RPC call (such as `eth_estimateGas`). Are the `from`, `to` addresses correct?"
+                    ]
+                  }
```

Comment:
This also doesn't feel particularly helpful. E.g. OK, these are _options_, but we have a more specific error inside already. Can't we be more specific? E.g. if transaction halts, it either needs to have more gas or has an unrelated problem (it halts always). If transaction reverts, it always reverts. Two latter options seem to provide a non-exhaustive subset of "halt" error -- e.g. it could be used to represent information about halting during estimation.

Maybe it should've been two different errors: one for halt and one for revert, rather than umbrella?
Actually, I guess in this scenario we can provide a better user-facing description: we can explain that pre-run with maximum gas failed with <this particular kind of error>, meaning that gas estimation is not possible.

Reply:
Actually, do we need this wrapper at all? Can't we use `GasEstimation` component directly to spare a level of nesting?